### ISSUE TITLE
Currently broken on windows python 3.12?

### ISSUE THREAD

**Issue by BrianPugh** [RobertCraigie/pyright-python#219]
[In one of my repos, tests pass](https://github.com/BrianPugh/python-template/actions/runs/6421555938/job/17436186990?pr=69) for all OS's (linux, macos, windows) and all python versions (python3.8-3.12) *except* for windows-py312:

```
pyright..................................................................Failed
- hook id: pyright
- exit code: 1

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\runneradmin\.cache\pre-commit\repo8zg16k_g\py_env-python3.12\Scripts\pyright.EXE\__main__.py", line 7, in <module>
  File "C:\Users\runneradmin\.cache\pre-commit\repo8zg16k_g\py_env-python3.12\Lib\site-packages\pyright\cli.py", line 34, in entrypoint
    sys.exit(main(sys.argv[1:]))
             ^^^^^^^^^^^^^^^^^^
  File "C:\Users\runneradmin\.cache\pre-commit\repo8zg16k_g\py_env-python3.12\Lib\site-packages\pyright\cli.py", line 19, in main
    return run(*args, **kwargs).returncode
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\runneradmin\.cache\pre-commit\repo8zg16k_g\py_env-python3.12\Lib\site-packages\pyright\cli.py", line 25, in run
    pkg_dir = install_pyright(args, quiet=None)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\runneradmin\.cache\pre-commit\repo8zg16k_g\py_env-python3.12\Lib\site-packages\pyright\_utils.py", line 61, in install_pyright
    node.run(
  File "C:\Users\runneradmin\.cache\pre-commit\repo8zg16k_g\py_env-python3.12\Lib\site-packages\pyright\node.py", line 122, in run
    subprocess.run(node_args, env=env, **kwargs),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.0\x64\Lib\subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.0\x64\Lib\subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\hostedtoolcache\windows\Python\3.12.0\x64\Lib\subprocess.py", line 1538, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [WinError 193] %1 is not a valid Win32 application
```

Maybe something changed in subprocess on windows in python3.12? I haven't taken a closer look.

**Comment by lukebakken** [comment]
See https://github.com/RobertCraigie/pyright-python/pull/220

**Comment by RobertCraigie** [comment]
Thanks for the bug report and thanks for the PR @lukebakken! We do not currently have CI tests for windows so I'm working on adding them and then I'll test your PR.

**Comment by RobertCraigie** [comment]
This fix has been released in `v1.1.330.post0`
